### PR TITLE
completed 3 files

### PR DESCRIPTION
**Pull request by DiemBTran**
I took Sarah's advice from Week 4 Assessment for the 3rd JavaScript question, so I hope it's an improvement from last time. The 1st question was a fun challenge! I had to console.log a lot to solve that one. Please let me know anything I can improve on, even though this is my last assessment.

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -10,9 +10,24 @@ var people = [
 ]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is a president of the galaxy." "Arthus Dent is a radio employee."
 
+// a function that takes in an array of objects
+const sentenceMaker = (object) => {
+  //iterate thru object
```
Comment (sjproctor) [inline code-challenges.js:15]: While the name of the argument is arbitrary, it does set certain expectations. In this case what you are passing in is an array of objects. You actually can't iterate on an object.

File: code-challenges.js [hunk 2]
```diff
@@ -10,9 +10,24 @@ var people = [
...
+    //map through array of first and last names to capitalize
+    let capitalizedName = person.map( (name) => {
+          //concat string of slices with first index capitalized
+          return name[0].toUpperCase().concat(name.slice(1))
+        }).join(" ")
+          //for each value, string interpolate
+        return `${capitalizedName} is a ${value.occupation}.`
+  }).join(" ")
+  return sentence
+}
```
Comment (sjproctor) [inline code-challenges.js:29]: Logic looks great!

File: code-challenges.js [hunk 3]
```diff
@@ -21,7 +36,20 @@ var testingArray1 = [23, "Heyyyy!", 45, -9, 0, "Yo", false]
...
+const moduloThree = (array) => {
+  //filters for type of integer only
+  let numbersOnly = array.filter(value => {
+    return typeof value === "number"
+  })
+  // map thru and returm modulo 3 
+  return numbersOnly.map(number => {
+    return number % 3
+  })
+}
```
Comment (sjproctor) [inline code-challenges.js:50]: Nice job!

File: code-challenges.js [hunk 4]
```diff
@@ -30,3 +58,21 @@ var testingArray2 = [5, "Hola", 43, -34, "greetings", true]
...
+  return finalArr.filter((value, index) =>{
+    //only return values if they're not already present in finalarr
+    //use indexOf to find first instance of that value
+    //if indexOf value === current index, then current index is the first time that value is seen
+    //thus, keep that value only and no future instances of value
+    return finalArr.indexOf(value) === index
+  })
+
+}
+
```
Comment (sjproctor) [inline code-challenges.js:77]: Nice job! Looks very clean!

File: interview-questions.md [hunk 5]
```diff
@@ -8,43 +8,54 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 2. Which RESTful routes must always be passed params? Why?
 
   Your answer:
-
+    Show, Update, and Destroy must all be passed parameters because they target a specific query.
   Researched answer:
-
+    Edit must also be passed a parameter, as it is the intermediary before update occurs. The params select a specific query by its primary key, and that query is the one that the route will act upon.
```
Comment (sjproctor) [inline interview-questions.md:22]: Nice!

File: interview-questions.md [hunk 6]
```diff
@@ -8,43 +8,54 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+1. As a user, I can see all of the current tasks.
+2. As a user, I can add more tasks.
+3. As a user, I can delete tasks.
+4. As a user, I can update task descriptions.
+5. As a user, I can mark tasks as completed.
+6. As a user, I can categorize tasks based on type of work.
+7. As a user, I can somehow priortize the importanace of tasks.
+8. As a user, I can see when a task should be completed by.
+9. As a user, I can add a completion date for each task.
+10. As a user, I can click and view a specific task and its entire description.
```
Comment (sjproctor) [inline interview-questions.md:61]: Great job getting all the CRUD actions in there!

File: rails-commenting.rb [hunk 7]
```diff
@@ -6,25 +6,25 @@
...
 
   def show
-    # 3)
+    # 3) We are creating an instance variable @post, which is assigned a single BlogPost instance with the query selector .find. The argument within .find selects a query with the given URL's parameter. When method show is invoked in the route, it will display the single BlogPost with that parameter :id.
     @post = BlogPost.find(params[:id])
   end
 
-  # 4)
+  # 4) The controller method new represents a midway point between Create and Read in CRUD. It is the Rails convention to display a form to the user to input.
   def new
```
Comment (sjproctor) [inline rails-commenting.rb:22]: Nice answer.

**Review by sjproctor — APPROVED** [review]
Great job, Diem! Way to finish really strong. Your logical approach to code keeps getting better. Keep it up.

File: code-challenges.js (hunk 1 above)
Comment (DiemBTran) [reply code-challenges.js:15]: Thank you for clarifying that! You are right, it is an array of objects.

**Comment by DiemBTran** [comment]
Thank you as always, Sarah! This is my final assessment merge :'( Thank you for everything

**Pull request merged**